### PR TITLE
ESSOptimizer: Check inputs and check stopping criteria more frequently

### PR DESCRIPTION
**Pull request by dweindl**
* Ensure we have some stopping criterion to not run infinitely long
* Avoid exceeding stopping criteria by checking them more frequently

### CODE REVIEW

File: pypesto/optimize/ess/ess.py [hunk 1]
```diff
@@ -122,6 +122,18 @@ def __init__(
...
+            # in this case, we'd run forever
+            raise ValueError(
+                "Either `max_iter`, `max_eval` or `max_walltime_s` have to be provided."
+            )
+        if max_eval is None:
+            max_eval = np.inf
+        if max_walltime_s is None:
+            max_walltime_s = np.inf
+        if max_iter is None:
+            max_iter = np.inf
```
Comment (PaulJonasJost) [inline pypesto/optimize/ess/ess.py:135]: Can we have something like sensible default values somewhat? at least for max_iter?

**Review by PaulJonasJost — APPROVED** [review]

File: pypesto/optimize/ess/ess.py (hunk 1 above)
Comment (dweindl) [reply pypesto/optimize/ess/ess.py:135]: What is a sensible default value?

Comment (PaulJonasJost) [reply pypesto/optimize/ess/ess.py:135]: I would not necessarily know, which was why I was asking :D but for general optimisers I think we have like a 1000 steps? 🤔 If you cannot think of one, also fine, would just think that having some kind of sensible default would be nicer than throwing an error.

Comment (dweindl) [reply pypesto/optimize/ess/ess.py:135]: The point here was to have at least one exit criterion. If I want an optimization to run for 1h, and it manages to do more than 1000 iterations then that's fine. I think it doesn't hurt to think about what one wants to have as exit criterion :sweat_smile:.

Comment (PaulJonasJost) [reply pypesto/optimize/ess/ess.py:135]: also fine I guess 😅

**Pull request merged**